src/meta-srv/src/service/admin/maintenance.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use common_meta::key::MAINTENANCE_KEY;
+use common_meta::rpc::store::PutRequest;
+use snafu::{OptionExt, ResultExt};
+use tonic::codegen::http;
+
+use crate::error::{self, Result};
+use crate::service::admin::HttpHandler;
+use crate::service::store::kv::KvStoreRef;
+pub struct MaintenanceHandler {
+    pub kv_store: KvStoreRef,
+}
+
+#[async_trait::async_trait]
+impl HttpHandler for MaintenanceHandler {
+    async fn handle(
+        &self,
+        _: &str,
+        params: &HashMap<String, String>,
+    ) -> Result<http::Response<String>> {
+        let switch_on = params
+            .get("switch_on")
+            .map(|on| on.parse::<bool>())
+            .context(error::MissingRequiredParameterSnafu { param: "switch-on" })?
+            .context(error::ParseBoolSnafu {
+                err_msg: "`switch_on` is not a valid bool",
+            })?;
+
+        let req = PutRequest {
+            key: MAINTENANCE_KEY.to_string().into_bytes(),
+            value: vec![],
+            prev_kv: false,
+        };
+
+        if switch_on {
+            self.kv_store.put(req).await?;
```

Comment:
Better to store (and clear) the maintenance mode value in the `in_memory` store, too. 

What's more, metasrv should retrieve the the maintenance mode value from `kv_store` to initialize it in the `in_memory` store, in `Metasrv::try_start` method. Because the metasrv itself could restart during the maintenance.

Reply:
I was ready to do like this before, but i find that the annotation indicates  in_memory will not be persist.Because  I was afraid to ambiguity in the annotation, i only store in kv_store

Reply:
The `in_memory` store is indeed not being persistent. That's why we have to retrieve the maintenance mode value from kv_store and set it in `in_memory` store when metasrv is started.